resource/generation/parser/parser.go

```diff
@@ -114,43 +113,35 @@ func ParseStructs(pkg *types.Package) ([]Struct, error) {
 	return parsedStructs, nil
 }
 
-func structMethods(s types.Type) []*types.Selection {
-	var methods []*types.Selection
-
-	// Need to iterate over the type and its pointer type because
-	// a method can use either as a receiver e.g. (a *app) or (a app)
-	for _, t := range []types.Type{s, types.NewPointer(s)} {
-		methodSet := types.NewMethodSet(t)
-		if methodSet.Len() == 0 {
-			continue
-		}
-
-		for method := range methodSet.Methods() {
-			methods = append(methods, method)
-		}
+func HasInterface(pkg *types.Package, s Struct, methodName string) bool {
```

Comment:
parameter name should be updated 
```suggestion
func HasInterface(pkg *types.Package, s Struct, interfaceName string) bool {
```